firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -228,12 +274,30 @@ public synchronized void onActivityResumed(Activity activity) {
     // Log the app start trace in a non-main thread.
     executorService.execute(this::logAppStartTrace);
 
-    if (isRegisteredForLifecycleCallbacks) {
+    if (!isExperimentTTIDEnabled && isRegisteredForLifecycleCallbacks) {
```

Comment:
What happens if the flag dynamically changes from No to YES (or) vice versa? Would that cause a situation where we try to unregister for callbacks when we have not registered or the other way about?

Reply:
Good question! Since this flag is only read once in the beginning of `AppStartTrace#onActivityResumed`, it's not a problem.
> Scenario 1: experiment disabled, an activity resumed, experiment enabled, another activity resumed.

After the first resume, the unregistering would be called, so the 2nd activity resume would not do anything. TTID isn't recorded for this app start, it will only start working next app start.

> Scenario 2: experiment enabled, an activity resumed, experiment enabled, another activity resumed.

After the first resume, the `AppStartTrace.onResume` variable would be set, so even if the 2nd activity resume triggers, nothing happens because this method will return immediately without sending anything else.

Reply:
Sounds good.

For the first scenario - I assume that experiment is disabled?

In the second scenario - I assume that would mean experiment is enabled?

Reply:
Yes, these scenarios show what happens if the app starts with 1 state, then changes after app start.